examples/plot_knockoff_aggregation.py

```diff
@@ -37,142 +38,146 @@
+        )
 
+        fdp_eval, power_eval = fdp_power(eval_selection, non_zero_index)
 
-fdps_mx = []
-fdps_pval = []
-fdps_eval = []
-powers_mx = []
-powers_pval = []
-powers_eval = []
+        return fdp_mx, fdp_pval, fdp_eval, power_mx, power_pval, power_eval
 
-for seed in seed_list:
-    fdp_mx, fdp_pval, fdp_eval, power_mx, power_pval, power_eval = single_run(
-        n_subjects, n_clusters, rho, sparsity, fdr, n_bootstraps, n_jobs, seed=seed
-    )
-    fdps_mx.append(fdp_mx)
-    fdps_pval.append(fdp_pval)
-    fdps_eval.append(fdp_eval)
-
-    powers_mx.append(fdp_mx)
-    powers_pval.append(power_pval)
-    powers_eval.append(power_eval)
-
-# Plot FDP and Power distributions
-
-fdps = [fdps_mx, fdps_pval, fdps_eval]
-powers = [powers_mx, powers_pval, powers_eval]
-
-
-def plot_results(bounds, fdr, nsubjects, n_clusters, rho, power=False):
-    plt.figure(figsize=(10, 10), layout="constrained")
-    for nb in range(len(bounds)):
-        for i in range(len(bounds[nb])):
-            y = bounds[nb][i]
-            x = np.random.normal(nb + 1, 0.05)
-            plt.scatter(x, y, alpha=0.65, c="blue")
-
-    plt.boxplot(bounds, sym="")
-    if power:
-        plt.xticks(
-            [1, 2, 3],
-            ["MX Knockoffs", "Quantile aggregation", "e-values aggregation"],
-            rotation=45,
-            ha="right",
+    parallel = Parallel(n_jobs, verbose=joblib_verbose)
+    results = parallel(
+        delayed(single_run)(
+            n_subjects, n_clusters, rho, sparsity, fdr, n_bootstraps, seed=seed
         )
-        plt.title(f"FDR = {fdr}, n = {nsubjects}, p = {n_clusters}, rho = {rho}")
-        plt.ylabel("Empirical Power")
-
-    else:
-        plt.hlines(fdr, xmin=0.5, xmax=3.5, label="Requested FDR control", color="red")
-        plt.xticks(
-            [1, 2, 3],
-            ["MX Knockoffs", "Quantile aggregation", "e-values aggregation"],
-            rotation=45,
-            ha="right",
-        )
-        plt.title(f"FDR = {fdr}, n = {nsubjects}, p = {n_clusters}, rho = {rho}")
-        plt.ylabel("Empirical FDP")
-        plt.legend(loc="best")
-
-    plt.show()
-
+        for seed in seed_list
+    )
 
-plot_results(fdps, fdr, n_subjects, n_clusters, rho)
-plot_results(powers, fdr, n_subjects, n_clusters, rho, power=True)
+    fdps_mx = []
+    fdps_pval = []
+    fdps_eval = []
+    powers_mx = []
+    powers_pval = []
+    powers_eval = []
+    for result in results:
+        fdp_mx, fdp_pval, fdp_eval, power_mx, power_pval, power_eval = result
+        fdps_mx.append(fdp_mx)
+        fdps_pval.append(fdp_pval)
+        fdps_eval.append(fdp_eval)
+
+        powers_mx.append(fdp_mx)
+        powers_pval.append(power_pval)
+        powers_eval.append(power_eval)
+
+    # Plot FDP and Power distributions
+
+    fdps = [fdps_mx, fdps_pval, fdps_eval]
+    powers = [powers_mx, powers_pval, powers_eval]
+
+    def plot_results(bounds, fdr, nsubjects, n_clusters, rho, power=False):
```

Comment:
Also this function should be defined outside of other functions imho.

Reply:
I just created a big function to show the different results depending on the n_sample.
Do you want to keep the two figures or only the one which is close to the original one?